decorator.rb

```diff
@@ -0,0 +1,10 @@
+class Decorator < Nameable
+  def initialize(nameable)
```

Comment:
- [ ] You will need to declare the `attr_accessor :nameable` so that it could be accessed by the correct_name method.